python/teams_ai/moderator/moderator.py

```diff
@@ -0,0 +1,38 @@
+"""
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the MIT License.
+"""
+
+from abc import ABC, abstractclassmethod
+from typing import Optional
+
+class Moderator(ABC):
+    """
+    interface implemented by all moderators
+    """
+
+    @abstractclassmethod
+    async def review_prompt(input: str) -> Optional[Plan]:
+        """
+        Reviews user input before it's sent to the planner.
+
+        Args:
+            input (str): The user input.
```

Comment:
What Steven proposed seems to moderate the rendered prompt as well? Correct me if I misunderstand.

Reply:
My understanding to #203 is since the planner implementation now uses a built-int prompt manager, there's no way to get the final prompt anymore. So the moderator can only review user's input.
But the interface doesn't care what's the content of input, maybe I can update the comment to avoid confusion.